test/integration/integration_test.go

```diff
@@ -38,6 +39,8 @@ const (
 	TagIncrementalBuildScripts                = "test/sti-incremental-app-scripts"
 	TagIncrementalBuildScriptsNoSaveArtifacts = "test/sti-incremental-app-scripts-no-save-artifacts"
 	TagCleanLayeredBuildNoTar                 = "test/sti-fake-no-tar"
+	TagCleanBuildOnBuild                      = "test/sti-onbuild-trigger"
+	TagOnBuildTrigger                         = "test/sti-file-check-onbuild-trigger"
```

Comment:
again I think this goes away, you just need a TagCleanBuildOnBuild image and, if you want, a TagIncrementalBuildOnBuild image.  Those being the 2 output images you're going to produce with an sti build invocation that uses the "onbuild" builder image.